app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/AccountController.scala

```diff
@@ -70,4 +73,26 @@ class AccountController @Inject()(
     }
   }
 
+  def getExistingUndertaking: Action[AnyContent] = escAuthentication.async { implicit request =>
+    implicit val eori: EORI = request.eoriNumber
+    for {
+      undertakingOpt <- escService.retrieveUndertaking(eori)
+    } yield {
+      undertakingOpt match {
+        case Some(undertaking) if(isLeadEORI(undertaking, eori)) => Redirect(routes.AccountController.getAccountPage())  //if logged in as lead EORI, redirect to Account home page
+        case Some(undertaking) => Ok(existingUndertakingPage(undertaking.name, eori)) //if logged in as non-lead EORI, redirect to existing undertaking page
+        case None => Redirect(routes.AccountController.getAccountPage()) //if undertaking not present, then create undertaking
+      }
+    }
+  }
+
+  //checks if the logged in EORI is a lead
+  private  def isLeadEORI(undertaking: Undertaking, eori: EORI): Boolean = {
```

Comment:
NABD - but I'd make the "isLeadEori" a function of the undertaking which accepts are EORI as it'll likely be reused.